frontend/dockerfile/docs/syntax.md

```diff
@@ -25,6 +25,52 @@ incrementing the major component of a version and you may want to pin the image
 change in between releases on labs channel, the old versions are guaranteed to be backward compatible.
 
 
+## Linked copies `COPY --link`, `ADD --link`
+
+To use this flag set Dockerfile version to at least `1.4`.
+
+```
+# syntax=docker/dockerfile:1.4
+```
+
+Enabling this flag in `COPY` or `ADD` commands allows you to copy files with enhanced semantics where your files remain independent on their own layer and don't get invalidated when commands on previous layers are changed.
+
+When `--link` is used your source files are copied into an empty destination directory. That directory is turned into a layer that is linked on top of your previous state.
+
+```
+# syntax=docker/dockerfile:1.4
+FROM alpine
+COPY --link /foo /bar
+```
+
+Is equivalent of doing two builds:
+
+```
+FROM alpine
+```
+
+and
+
+```
+FROM scratch
+COPY /foo /bar
+```
+
+and merging all the layers of both images together.
```

Comment:
> merging all the layers

Does that mean that this produces a single layer (alpine + copy) ? Won't that result in the `alpine` layer no longer being shared? (I recall we tried to avoid that when implementing `docker build --squash`, where we kept the base-layer out of the squashing to allow it to still be shared).

```dockerfile
FROM alpine
COPY --link /foo /bar
```


This somewhat implies that;

```dockerfile
FROM alpine
COPY --link /foo /bar
COPY --link /bar /baz
```

Would result in a single layer (alpine + copy + copy) (is that correct?)

Reply:
Side note, I'm working on some user docs for the lower-level MergeOp. We can consider updating these Dockerfile docs to mention those ones once I'm done, which might help clarify questions like this that have to do with finer-grain details.